auth/api/auth/v2/api.go

```diff
@@ -0,0 +1,162 @@
+func (r Wrapper) HandleTokenRequest(ctx context.Context, request HandleTokenRequestRequestObject) (HandleTokenRequestResponseObject, error) {
+	switch request.Body.GrantType {
+	case "authorization_code":
+		// Options:
+		// - OpenID4VCI
+		// - OpenID4VP, vp_token is sent in Token Response
+	case "vp_token":
+		// Options:
+		// - service-to-service vp_token flow
+	case "urn:ietf:params:oauth:grant-type:pre-authorized_code":
+		// Options:
+		// - OpenID4VCI
+	default:
+		return nil, openid4vci.Error{
+			Code:       openid4vci.InvalidRequest,
+			StatusCode: http.StatusBadRequest,
+			//Description: "invalid grant type",
+		}
+	}
+
+	// TODO: Handle?
+	//scope, err := handler(request.Body.AdditionalProperties)
+	//if err != nil {
+	//	return nil, err
+	//}
+	// TODO: Generate access token with scope
+	return HandleTokenRequest200JSONResponse(TokenResponse{
+		AccessToken: "",
+	}), nil
+}
+
+// HandleAuthorizeRequest handles calls to the authorization endpoint for starting an authorization code flow.
+func (r Wrapper) HandleAuthorizeRequest(ctx context.Context, request HandleAuthorizeRequestRequestObject) (HandleAuthorizeRequestResponseObject, error) {
+	// Create session object to be passed to handler
+	session := &Session{
+		// TODO: Validate client ID
+		ClientID: request.Body.ClientId,
+	}
+	// TODO: Validate scope?
+	if request.Body.Scope != nil {
+		session.Scope = *request.Body.Scope
+	}
+	if request.Body.State != nil {
+		session.ClientState = *request.Body.State
+	}
+	// TODO: Validate redirect URI
+	if request.Body.RedirectUri != nil {
+		// TODO: Validate according to https://datatracker.ietf.org/doc/html/rfc6749#section-3.1.2
+		session.RedirectURI = *request.Body.RedirectUri
+	} else {
+		// TODO: Spec says that the redirect URI is optional, but it's not clear what to do if it's not provided.
+		//       See https://datatracker.ietf.org/doc/html/rfc6749#section-4.1.1
+		return nil, errors.New("missing redirect URI")
+	}
+
+	switch request.Body.ResponseType {
+	case "code":
+		// Options:
+		// - Regular authorization code flow for EHR data access through access token, authentication of end-user using OpenID4VP.
+		// - OpenID4VCI; authorization code flow for credential issuance to (end-user) wallet
+		// - OpenID4VP, vp_token is sent in Token Response; authorization code flow for presentation exchange (not required a.t.m.)
+		// TODO: Switch on parameters to right flow
+	case "vp_token":
+		// Options:
+		// - OpenID4VP flow, vp_token is sent in Authorization Response
+		// TODO: Check parameters for right flow
+		// TODO: Do we actually need this? (probably not)
+	case "vp_token id_token":
+		// Options:
+		// - OpenID4VP+SIOP flow, vp_token is sent in Authorization Response
+		// TODO: Check parameters for right flow
+	default:
+		// TODO: This should be a redirect?
+		return nil, openid4vci.Error{
+			Code:       openid4vci.InvalidRequest,
```

Comment:
should be error code
[4.1.2.1 Error Response](https://datatracker.ietf.org/doc/html/rfc6749#section-4.1.2.1)
```
unsupported_response_type
      The authorization server does not support obtaining an
      authorization code using this method.
```
or add issue as reminder to validate errors?

Reply:
Added TODO for errors; they shouldn't reside in the openid4vci package